anta/runner.py

```diff
@@ -56,65 +70,57 @@ def log_cache_statistics(devices: list[AntaDevice]) -> None:
-        tests: tests to run against devices. None means all tests.
         tags: Tags to filter devices from the inventory.
-        established_only: Include only established device(s).
-    """
-    limits = resource.getrlimit(resource.RLIMIT_NOFILE)
-    logger.debug("Initial limit numbers for open file descriptors for the current ANTA process: Soft Limit: %s | Hard Limit: %s", limits[0], limits[1])
-    nofile = __NOFILE__ if limits[1] > __NOFILE__ else limits[1]
-    logger.debug("Setting soft limit for open file descriptors for the current ANTA process to %s", nofile)
-    resource.setrlimit(resource.RLIMIT_NOFILE, (nofile, limits[1]))
-    limits = resource.getrlimit(resource.RLIMIT_NOFILE)
+        devices: Devices on which to run tests. None means all devices.
 
-    if not catalog.tests:
-        logger.info("The list of tests is empty, exiting")
-        return
+    Returns
+    -------
+        AntaInventory | None: The filtered inventory or None if there are no devices to run tests on.
+    """
     if len(inventory) == 0:
         logger.info("The inventory is empty, exiting")
-        return
+        return None
 
-    # Filter the inventory based on tags and devices parameters
-    selected_inventory = inventory.get_inventory(
-        tags=tags,
-        devices=devices,
-    )
+    # Filter the inventory based on the CLI provided tags and devices if any
+    selected_inventory = inventory.get_inventory(tags=tags, devices=devices) if tags or devices else inventory
+
+    # Connect to the devices
     await selected_inventory.connect_inventory()
 
     # Remove devices that are unreachable
     inventory = selected_inventory.get_inventory(established_only=established_only)
 
+    # If there are no devices in the inventory after filtering, exit
     if not inventory.devices:
         msg = f'No reachable device {f"matching the tags {tags} " if tags else ""}was found.{f" Selected devices: {devices} " if devices is not None else ""}'
         logger.warning(msg)
-        return
-    coros = []
+        return None
+
+    return selected_inventory
 
-    # Select the tests from the catalog
-    if tests:
-        catalog = AntaCatalog(catalog.get_tests_by_names(tests))
+
+async def prepare_tests(inventory: AntaInventory, catalog: AntaCatalog, tests: set[str] | None, tags: set[str] | None) -> set[AntaTestRunner] | None:
+    """Prepare the tests to run.
+
+    Args:
+    ----
+        inventory: AntaInventory object that includes the device(s).
+        catalog: AntaCatalog object that includes the list of tests.
+        tests: Tests to run against devices. None means all tests.
+        tags: Tags to filter devices from the inventory.
+
+    Returns
+    -------
+        set[AntaTestRunner] | None: The selected tests to run or None if there are no tests to run.
+    """
+    # Build indexes for the catalog. If there are CLI provided tests, filter the indexes based on these tests
```

Comment:
test are not necessarily provided by CLI
```suggestion
    # Build indexes for the catalog. If 'tests' is set. filter the indexes based on these tests
```

Reply:
Fixed in 97876f0f197057a008ebf95d8dae1c6169b71650